contracts/cw721-expiration/src/query.rs

```diff
@@ -1,296 +1,347 @@
+            )?),
         }
     }
 
-    pub fn minter(&self, deps: Deps) -> StdResult<MinterResponse> {
-        self.base_contract.minter(deps)
-    }
-
-    pub fn ownership(deps: Deps) -> StdResult<cw_ownable::Ownership<Addr>> {
-        cw_ownable::get_ownership(deps.storage)
-    }
-}
-
-// queries
-impl<'a> Cw721ExpirationContract<'a> {
-    pub fn contract_info(&self, deps: Deps) -> StdResult<ContractInfoResponse> {
-        self.base_contract.contract_info(deps)
-    }
-
-    pub fn num_tokens(&self, deps: Deps) -> StdResult<NumTokensResponse> {
-        self.base_contract.num_tokens(deps)
-    }
-
-    pub fn nft_info(
+    pub fn query_nft_info_include_expired_nft(
         &self,
         deps: Deps,
         env: Env,
         token_id: String,
-        include_invalid: bool,
-    ) -> Result<NftInfoResponse<Extension>, ContractError> {
-        if !include_invalid {
-            self.assert_valid_nft(deps, &env, token_id.as_str())?;
+        include_expired_nft: bool,
+    ) -> Result<NftInfoResponse<TMetadataExtension>, ContractError> {
+        if !include_expired_nft {
+            self.assert_nft_expired(deps, &env, token_id.as_str())?;
         }
-        Ok(self.base_contract.nft_info(deps, token_id)?)
+        Ok(self.base_contract.query_nft_info(deps, env, token_id)?)
     }
 
-    pub fn owner_of(
+    pub fn query_owner_of_include_expired_nft(
         &self,
         deps: Deps,
         env: Env,
         token_id: String,
-        include_expired: bool,
-        include_invalid: bool,
+        include_expired_approval: bool,
+        include_expired_nft: bool,
     ) -> Result<OwnerOfResponse, ContractError> {
-        if !include_invalid {
-            self.assert_valid_nft(deps, &env, token_id.as_str())?;
+        if !include_expired_nft {
+            self.assert_nft_expired(deps, &env, token_id.as_str())?;
         }
         Ok(self
             .base_contract
-            .owner_of(deps, env, token_id, include_expired)?)
+            .query_owner_of(deps, env, token_id, include_expired_approval)?)
     }
 
-    /// operator returns the approval status of an operator for a given owner if exists
-    pub fn operator(
-        &self,
-        deps: Deps,
-        env: Env,
-        owner: String,
-        operator: String,
-        include_expired: bool,
-    ) -> StdResult<OperatorResponse> {
-        self.base_contract
-            .operator(deps, env, owner, operator, include_expired)
-    }
-
-    /// operators returns all operators owner given access to
-    pub fn operators(
-        &self,
-        deps: Deps,
-        env: Env,
-        owner: String,
-        include_expired: bool,
-        start_after: Option<String>,
-        limit: Option<u32>,
-    ) -> StdResult<OperatorsResponse> {
-        self.base_contract
-            .operators(deps, env, owner, include_expired, start_after, limit)
-    }
-
-    pub fn approval(
+    pub fn query_approval_include_expired_nft(
```

Comment:
How come we add "query" in the name? Before they just say "approval" so I'm not sure if naming convention necessitates that we put "query" in the name or not?

Reply:
again, like above diff here, just show it propery. all general cw721 queries are moved into package. All contract specific logic are kept here.